pkg/engine/virtualmachines.go

```diff
@@ -361,6 +363,8 @@ func createAgentAvailabilitySetVM(cs *api.ContainerService, profile *api.AgentPo
 		Tags: tags,
 	}
 
+	addCustomTagsToVM(cs.Properties.MasterProfile.CustomVMTags, &virtualMachine)
```

Comment:
this should be `addCustomTagsToVM(profile.CustomVMTags, &virtualMachine)`

Reply:
Sure! I didn't noticed this was already been instantiated! Thanks for the suggestion!

Reply:
Hey @CecileRobertMichon , can you explain why I have to use the profile here, instead of the cs? I'm just curious cause I'm getting the right arm templates when I use the cs parameter, but not getting the tags when I change it to the profile. Thanks a lot!

Reply:
This specific function is to add tags to the agent VMs, so you want the tags in the agent pool profile part of the apimodel. `cs.Properties.MasterProfile` is the master profile of the cluster, so that should only be used for master VMs. In this context, `profile` is the agent pool profile that we're looping on so that's what we want.

If you're not getting the tags while testing maybe that's because you didn't add tags to the agent pool profiles or maybe you're checking the wrong VM?

Reply:
I was not converting to vlabs correctly. Thanks a lots for the comments